src/commands/importFunctionApp/importFunctionApp.ts

```diff
@@ -0,0 +1,319 @@
+    const operationPolicy = [{
+        policies: [
+            {
+                inbound: [
+                    { base: null },
+                    {
+                        "set-backend-service": [
+                            {
+                                _attr: {
+                                    id: "apim-generated-policy",
+                                    "backend-id": backendId
+                                }
+                            }
+                        ]
+                    }
+                ]
+            },
+            { backend: [{ base: null }] },
+            { outbound: [{ base: null }] },
+            { "on-error": [{ base: null }] }
+        ]
+    }];
+
+    return xml(operationPolicy);
+}
+
+// Create backend for the function app on api service
+async function setAppBackendEntity(backendId: string, appName: string, appPath: string, BackendCredentials: BackendCredentialsContract, node: ApisTreeItem | ApiTreeItem): Promise<void> {
+    const nbackend: BackendContract = {
+        description: `${appName}`,
+        resourceId: `${node.root.environment.resourceManagerEndpointUrl}/subscription/${node.root.subscriptionId}/resourceGroups/${node.root.resourceGroupName}/providers/Microsoft.Web/sites/${appName}`,
+        url: appPath,
+        id: backendId,
+        name: backendId,
+        protocol: "http",
+        credentials: BackendCredentials
+    };
+    await node.root.client.backend.createOrUpdate(node.root.resourceGroupName, node.root.serviceName, backendId, nbackend);
+}
+
+// Create secret named value for function app hostkey
+async function createPropertyItem(node: ApisTreeItem | ApiTreeItem, propertyId: string, hostKey: string): Promise<void> {
+    const propertyContract: PropertyContract = {
+        displayName: propertyId,
+        value: hostKey,
+        tags: ["key", "function", "auto"],
+        secret: true
+    };
+    await node.root.client.property.createOrUpdate(node.root.resourceGroupName, node.root.serviceName, propertyId, propertyContract);
+}
+
+// create new operation for each function and method
+function getNewOperation(apiId: string, method: string, name: string, route: string, displayName: string): OperationContract {
+    const operationId = apiUtil.displayNameToIdentifier(name);
+    const cleanUrl = parseUrlTemplate(route);
+    return {
+        id: `${apiId}/operations/${operationId}`,
+        name: operationId,
+        displayName: displayName,
+        method: method,
+        urlTemplate: cleanUrl.urlTemplate,
+        description: "",
+        templateParameters: cleanUrl.parameters
+    };
+}
+
+function getFunctionAppBase(invokeUrlTemplate: string, route: string): string {
+    const uri = invokeUrlTemplate.substr(0, invokeUrlTemplate.length - route.length);
+    return uri.endsWith("/") ? uri.substr(0, uri.length - 1) : uri;
+}
+
+async function createApiFromFunctionApp(apiId: string, funcApp: Site, apiName: string): Promise<ApiContract> {
```

Comment:
name function as construct or build instead of create as we dont create the API Yet?